Update script.js

OlyaEl opened the pull request:
Сделала копию исходного массива. Используя шаблон создала карточку с наполнением из массива. Вывела в интерфейс новый массив-копию.

OlyaEl committed:
Update script.js

Сделала копию исходного массива. Используя шаблон создала карточку с наполнением из массива. Вывела в интерфейс новый массив-копию.

PolinaShneider commented on script.js:
```diff
@@ -1,86 +1,120 @@
+const descriptionCard = cardTemplate.content.querySelector('p');
+const tagsContainer = cardTemplate.content.querySelector('.tags');
+const priceCard = cardTemplate.content.querySelector('span');
+const imageCard = cardTemplate.content.querySelector('img');
```
Мне не очень нравится, что у тебя части шаблона вынесены в глобальные переменные. Тебе ж по сути нужно заполнить контентом копию шаблона для конкретной карточки. Давай перепишем код функции вот так

```
function createShopCard(cardItem) {
    const { title, description, tags, price, img } = cardItem;
    const card = cardTemplate.content.cloneNode(true);

    card.querySelector("h1").textContent = title;
    card.querySelector("p").textContent = description;
    card.querySelector(".tags").textContent = tags;
    card.querySelector("span").textContent = price;
    card.querySelector("img").src = img;

    return card;

}
```

PolinaShneider commented on script.js:
```diff
@@ -1,86 +1,120 @@
+
+}
+
+showCards(shopContent);
```
Тут ты передаешь массив в качестве параметра, но в функции он у тебя не описан и ты его не используешь. Если оставишь просто `showCards();`, эффект будет тот же

PolinaShneider commented on script.js:
```diff
@@ -1,86 +1,120 @@
 ];
+
+
+const shopContent = items.slice();
```
Не обязательно создавать копию. Ты же никак не меняешь исходный массив и используешь его только для чтения

OlyaEl committed:
Update script.js

Внесла необходимые изменения: убрала глобальные переменные, копию массива, параметры функции.
